### PR TITLE
fix airgap deploy

### PR DESCRIPTION
**Pull request by simonfelding**
# Description

added better support for airgap mode 'exists' in rke2.yml
Part of splitting up https://github.com/lablabs/ansible-role-rke2/pull/203

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Role code (GitHub Actions Workflow, Documentation etc.)

## How Has This Been Tested?
by rolling it out on an airgapped system multiple times

### CODE REVIEW

File: tasks/rke2.yml [hunk 1]
```diff
@@ -5,53 +5,102 @@
...
 
-- name: Download RKE2 checksum and artifacts
-  when: rke2_airgap_mode and rke2_airgap_implementation == 'download'
+- name: Download RKE2 checksum and artifacts (try to download remotely first - fallback to local download and push)
+  when:
+    - rke2_airgap_mode
+    - rke2_airgap_implementation == 'download'
   block:
-  - name: Download sha256 checksum file
+  - name: "Download sha256 checksum file"
```
Comment (MonolithProjects) [inline tasks/rke2.yml:34]: Quotation marks are not needed here

File: tasks/rke2.yml [hunk 2]
```diff
@@ -166,4 +272,6 @@
     owner: root
     group: root
     mode: 0644
-  when: rke2_environment_options is defined and rke2_environment_options|length > 0
+  when:
+    - rke2_environment_options is defined
+    - rke2_environment_options|length > 0
```
Comment (MonolithProjects) [inline tasks/rke2.yml:277]: Please remove the empty line

**Review by MonolithProjects — CHANGES_REQUESTED** [review]
Just small remarks. Otherwise it looks OK. Also please resolve the conflicts.

File: tasks/rke2.yml (hunk 1 above)
Comment (simonfelding) [reply tasks/rke2.yml]: ok

File: tasks/rke2.yml (hunk 2 above)
Comment (simonfelding) [reply tasks/rke2.yml]: thanks for the link! it fails because there is no empty line at the end of file - so I added one.

**Comment by simonfelding** [comment]
oops i messed up a bit sorry, i'll fix soon

**Comment by Raboo** [comment]
I'm eagerly looking forward for this PR to be merged. Currently the airgap install doesn't work for me, this PR appears to be the salvation.

**Comment by simonfelding** [comment]
Thanks for saying that! I'll get it done ASAP! It's so encouraging to hear.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Raboo** [comment]
I can try if I manage. Whats the status on your PR? What is missing or not working?

**Comment by Raboo** [comment]
@MonolithProjects Hi, what is needed for this PR to be merged?

**Comment by simonfelding** [comment]
@Raboo I thought I had messed up something when I did that force push but it seems to be fine? idk. lgtm.

**Review by MonolithProjects — APPROVED** [review]
LGTM

**Pull request merged**